pkg/procmatch/procmatch.go

```diff
@@ -0,0 +1,34 @@
+package procmatch
+
+//go:generate go run gen/generate_catalog.go
+
+// Matcher interface implements the Match method that takes a command line string and returns a potential integration for this command line string
+type Matcher interface {
+	Match(cmdline string) string // Return the name of the integration
+}
+
+// Integration represents an integration
+type Integration struct {
+	Name       string   // Name of the integration
+	Signatures []string // Signatures of the integration's command line processes
+}
+
+// IntegrationCatalog represents a list of Integrations
+type IntegrationCatalog []Integration
+
+// Default matcher
+var defaultGraphMatcher Matcher
+
+func init() {
+	var err error
+	defaultGraphMatcher, err = NewMatcher(DefaultCatalog)
+
+	if err != nil {
+		panic(err)
```

Comment:
Adding a panic seems very risky, especially for a customer-side application. That will mean if there's some odd bug introduced we could potentially block the whole Agent from starting when it's just the process matching that's broken.

I'd prefer this to log a big error and then have some handling of an uninitialized `Matcher`

Reply:
We could remove the `init` function and the `Match` and provide a `NewDefault` that would use the default catalog:

```golang
func NewDefault() (Matcher, error) { return NewMatcher(DefaultCatalog) }
```

This way the error handling would be in the "library client" side